src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/PreferHashDataOverComputeHashTests.cs

```diff
@@ -0,0 +1,1487 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.PreferHashDataOverComputeHashAnalyzer,
+    Microsoft.NetCore.CSharp.Analyzers.Performance.CSharpPreferHashDataOverComputeHashFixer>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.PreferHashDataOverComputeHashAnalyzer,
+    Microsoft.NetCore.VisualBasic.Analyzers.Performance.BasicPreferHashDataOverComputeHashFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Performance.UnitTests
+{
+    public class PreferHashDataOverComputeHashTests
+    {
+        private const string HashTypeMD5 = "MD5";
+        private const string HashTypeSHA1 = "SHA1";
+        private const string HashTypeSHA256 = "SHA256";
+        private const string HashTypeSHA384 = "SHA384";
+        private const string HashTypeSHA512 = "SHA512";
+
+        [Fact]
+        public async Task CSharpBailOutNoFixCase()
+        {
+            await TestWithType(HashTypeMD5);
+            await TestWithType(HashTypeSHA1);
+            await TestWithType(HashTypeSHA256);
+            await TestWithType(HashTypeSHA384);
+            await TestWithType(HashTypeSHA512);
+
+            static async Task TestWithType(string hashType)
+            {
+                string csInput = $@"
+using System;
+using System.Security.Cryptography;
+
+public class Test
+{{
+    public static void TestMethod({hashType} hash)
+    {{
+        var buffer = new byte[1024];
+        int aboveLine = 20;
+        byte[] digest = hash.ComputeHash(buffer);
```

Comment:
Yes, I strictly followed the rules as described in the issue

Reply:
So, this would be the expected result? 
I think it's okay
```cs
               string csInput = $@"
using System;
using System.Security.Cryptography;
public class Test
{{
    public static void TestMethod({hashType} hash)
    {{
        var buffer = new byte[1024];
        int aboveLine = 20;
        byte[] digest = {{|#0:hash.ComputeHash({{|#1:buffer|}})|}}
        int belowLine = 10;
    }}
}}
";
                string csFix = $@"
using System;
using System.Security.Cryptography;
public class Test
{{
    public static void TestMethod({hashType} hash)
    {{
        var buffer = new byte[1024];
        int aboveLine = 20;
        byte[] digest = {hashType}.HashData(buffer);
        int belowLine = 10;
    }}
}}
";
```

Reply:
so if there's no Create, I should not report? I'll update again tmr

Reply:
> * If the hashType is just `HashAlgorithm` then it's polymorphic and there's no fix.

The analyzer checks if the algorithm has the static HashData(...) method

> * Even if the hashType is one of the algorithms, they're not sealed, polymorphism could have been desired.
> * And even if not, there's data bleed between hash.TransformBlock and hash.ComputeHash, so the incoming object could have "intentionally corrupted" state that's supposed to mix in.

@bartonjs thanks for the clarification 

> so if there's no Create, I should not report? I'll update again tmr

@wzchua sorry about that and thanks for the update!

Reply:
> And even if not, there's data bleed between hash.TransformBlock and hash.ComputeHash, so the incoming object could have "intentionally corrupted" state that's supposed to mix in.

does this mean if there's a call to `hash.TransformBlock` and then a call to `hash.ComputeHash`, I shouldn't report?